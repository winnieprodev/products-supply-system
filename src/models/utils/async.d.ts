export interface Error {
  message: string;
}

export interface Async<T> {
  isLoading: boolean;
  error: Error | undefined;
  data: T | undefined;
}
