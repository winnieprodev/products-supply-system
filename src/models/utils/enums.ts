export enum WrapperMode {
  full = "full",
  singleLine = "singleLine",
  tiny = "tiny",
}
