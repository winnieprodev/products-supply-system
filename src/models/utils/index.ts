export * from "./async.d";
export * from "./enums";
