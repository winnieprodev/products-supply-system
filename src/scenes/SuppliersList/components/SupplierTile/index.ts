export { default as SupplierTile } from "./SupplierTile";
