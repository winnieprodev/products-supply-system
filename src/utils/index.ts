export * from "./routeUtils";
export * from "./supplierUtils";
export * from "./testUtils";
