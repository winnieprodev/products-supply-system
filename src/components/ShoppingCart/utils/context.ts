import { Async, Cart, ProductCartInfo } from "models";
import React from "react";
import { mockAsync, mockCart } from "utils";

interface ShoppingCartContextProps {
  cartId: Async<number>;
  cart: Async<Cart> | undefined;
  setCart: (cart: Async<Cart> | undefined) => void;
  setProduct: (product: ProductCartInfo) => void;
  removeProduct: (productId: string) => void;
}

export const EMPTY_CART = { products: [] };
const mockFunction = () => {};

export const ShoppingCartContext = React.createContext<ShoppingCartContextProps>(
  {
    cartId: mockAsync(0),
    cart: mockAsync(mockCart(0)),
    setCart: mockFunction,
    setProduct: mockFunction,
    removeProduct: mockFunction,
  },
);
