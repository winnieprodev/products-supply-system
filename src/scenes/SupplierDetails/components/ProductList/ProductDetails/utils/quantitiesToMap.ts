import { Map } from "immutable";
import { CartQuantity, QuantityType } from "models";

export const quantitiesToMap = (quantities: CartQuantity[]) => {
  return Map<QuantityType, number>(
    quantities.map(({ type, amount }) => [type, amount] as const) as Iterable<
      [QuantityType, number]
    >,
  );
};
