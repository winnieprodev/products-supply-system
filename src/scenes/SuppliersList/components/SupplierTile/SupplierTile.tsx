import { GridTile } from "components";
import { Supplier } from "models";
import React from "react";

interface OwnProps {
  onClick: () => void;
  item: Supplier;
}

const SupplierTile: React.FC<React.PropsWithChildren<OwnProps>> = ({
  onClick,
  item,
}) => {
  return <GridTile onClick={onClick} title={item.title} mono={item.mono} />;
};

export default React.memo(SupplierTile);
