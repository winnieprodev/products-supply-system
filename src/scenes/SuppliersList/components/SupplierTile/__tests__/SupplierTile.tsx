import { render } from "@testing-library/react";
import { mockSupplier } from "utils";

import SupplierTile from "../SupplierTile";

describe("SupplierTile", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot", () => {
      const { container } = render(
        <SupplierTile
          item={mockSupplier("id1")}
          onClick={jest.fn()}
        ></SupplierTile>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
