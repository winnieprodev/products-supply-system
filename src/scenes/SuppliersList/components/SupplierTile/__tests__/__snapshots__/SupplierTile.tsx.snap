// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SupplierTile Snapshot tests it matches the snapshot 1`] = `
.c2 {
  background: #F3F3F3;
  border-radius: 10px;
  position: relative;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  cursor: pointer;
}

.c0:hover .c1 {
  background: #EEEEEE;
}

.c3 {
  color: rgba(0,0,0,0.25);
  text-shadow: 0 5px 10px white;
  font-size: 6vw;
  font-weight: bold;
  text-align: center;
  line-height: calc(8vw + 80px);
}

.c4 {
  margin: 8px;
  text-align: center;
  font-size: 16px;
  font-weight: bold;
}

<div
  class="c0"
>
  <div
    class="c1 c2"
  >
    <div
      class="c3"
    >
      i
    </div>
  </div>
  <div
    class="c4"
  >
    id1
  </div>
</div>
`;
